src/OrchardCore.Modules/OrchardCore.Contents/Sitemaps/DefaultContentItemsQueryProvider.cs

```diff
@@ -64,15 +61,22 @@ public async Task<ContentItemsQueryResult> GetContentItemsAsync(ContentTypesSite
             query = query.Where(x => x.ContentType.IsIn(typesToIndex) && x.Published);
         }
 
-        return new ContentItemsQueryResult
+        query = query.OrderBy(x => x.CreatedUtc)
+            .ThenBy(x => x.Id);
+
+        if (take.HasValue && take > 0)
```

Comment:
This will be removed but eventually this can be written `if (take is not null and > 0)` ;)